utils/RESTWalker.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2016 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import io
+import jinja2
+import json
+from pprint import pprint as pp
+import re
+
+import os
+
+UTILSDIR = os.path.abspath(os.path.dirname(__file__))
+COMMONPREFIX = os.path.dirname(UTILSDIR)
+SOURCE_ROOTDIR = os.path.join(COMMONPREFIX, 'f5', 'bigip')
+TEMPLATEDIR = os.path.join(UTILSDIR, 'template_library')
+DEVICECONFDIR = os.path.join(UTILSDIR, 'device_configs')
+
+
+class Kindless(Exception):
+    pass
+
+
+def load_json_from_pathname(pathname):
+    fp = io.open(pathname)
+    obj = json.load(fp)
+    fp.close()
+    return obj
+
+
+def dump_json(obj, pathname):
+    fp = io.open(pathname, 'wb')
+    json.dump(obj, fp, indent=1)
+    fp.close()
+
+
+class TemplateEngine(object):
```

Comment:
This is puntable, but I'd like to see docstrings for the reasons why this thing exists. And possible info on the edge cases.